Lib9c/Action/Sweep.cs

```diff
@@ -0,0 +1,263 @@
+
+            if (!world.IsStageCleared && stageId > world.StageClearedId)
+            {
+                throw new InvalidStageException(
+                    $"{addressesHex}Aborted as the stage ({worldId}/{stageId}) is not cleared; " +
+                    $"cleared stage: {world.StageClearedId}"
+                );
+            }
+
+            if (worldId == GameConfig.MimisbrunnrWorldId)
+            {
+                throw new InvalidWorldException(
+                    $"{addressesHex} [{worldId}] can't execute sweep action.");
+            }
+
+            var materialItemSheet = sheets.GetSheet<MaterialItemSheet>();
+
+            // check ap
+            if (apStoneCount > 0)
+            {
+                if (apStoneCount > usableApStoneCount)
+                {
+                    throw new UsageLimitExceedException();
+                }
+
+                // use apStone
+                var row = materialItemSheet.Values.First(r => r.ItemSubType == ItemSubType.ApStone);
+                if (!avatarState.inventory.RemoveFungibleItem(row.ItemId, context.BlockIndex, count: apStoneCount))
+                {
+                    throw new NotEnoughMaterialException(
+                        $"{addressesHex}Aborted as the player has no enough material ({row.Id})");
+                }
+            }
+            else
+            {
+                if (avatarState.actionPoint < stageRow.CostAP)
+                {
+                    throw new NotEnoughActionPointException(
+                        $"{addressesHex}Aborted due to insufficient action point: " +
+                        $"{avatarState.actionPoint} < required cost : {stageRow.CostAP})"
+                    );
+                }
+            }
+
+            var gameConfigState = states.GetGameConfigState();
+            if (gameConfigState is null)
+            {
+                throw new FailedLoadStateException(
+                    $"{addressesHex}Aborted as the game config state was failed to load.");
+            }
+
+            // apply
+            var itemPlayCount = gameConfigState.ActionPointMax / stageRow.CostAP * apStoneCount;
+            var apPlayCount = avatarState.actionPoint / stageRow.CostAP;
+            var playCount = apPlayCount + itemPlayCount;
+
+            var stageWaveSheet = sheets.GetSheet<StageWaveSheet>();
+            UpdateMonsterMap(avatarState, stageWaveSheet);
+
+            var rewardItems = GetRewardItems(context.Random, playCount, stageRow, materialItemSheet);
+            UpdateInventory(avatarState, rewardItems);
+
+            var levelSheet = sheets.GetSheet<CharacterLevelSheet>();
+            var (level, exp) = GetLevelAndExp(levelSheet, avatarState, stageId, playCount);
+            UpdateExp(avatarState, level, exp);
+
+            return states
+                .SetState(inventoryAddress, avatarState.inventory.Serialize())
+                .SetState(worldInformationAddress, avatarState.worldInformation.Serialize())
```

Comment:
we can skip worldInformation serialize by migrationRequired from `TryGetAvatarStateV2`